packages/is-shallow-equal/index.js

```diff
@@ -0,0 +1,90 @@
+var isArray = Array.isArray,
+	keys = Object.keys;
+
+/**
+ * Returns true if the two objects are shallow equal, or false otherwise.
+ *
+ * @param {Object} a First object to compare.
+ * @param {Object} b Second object to compare.
+ *
+ * @return {Boolean} Whether the two objects are shallow equal.
+ */
+function isShallowEqualObjects( a, b ) {
+	if ( a === b ) {
+		return true;
+	}
+
+	var aKeys = keys( a ),
+		bKeys = keys( b ),
```

Comment:
I wonder if a `for in` loop could be better than relying on keys performance wise?

Reply:
I guess this one is better if the keys length is different?

Reply:
`Object.keys` is actually quite fast. Faster than `for in`: https://jsperf.com/object-keys-iteration/20

And yes, there's a shortcutting opportunity here that we can assume the two values are different if they have keys of unequal length. Also not just for shortcutting, but verifying in case the second object has more keys than the first that we don't miss out on considering those (`a = { foo: 1 }, b = { foo: 1, bar: 2 }`)